### PR TITLE
Added the VBO EBO VAO classes

### PR DESCRIPTION
**Pull request by MiliAxe**
This PR adds the corresponding EBO VBO VAO classes.

### CODE REVIEW

**Comment by Sephixum** [comment]
Thank you for the hard work on your part. It is indeed appreciated.
Though there is a recommendation on class declaration part, for functions that do not change member variables please use "const" keyword after parameters parenthesis.
It is worth adding if function does not throw exceptions use keyword "noexcept" since it does in fact reduce executable size.
I would recommend taking a look at the [this](https://stackoverflow.com/questions/10787766/when-should-i-really-use-noexcept)

File: include/vao.hpp [hunk 1]
```diff
@@ -20,6 +20,6 @@ class VAO {
   void bind();
   static void unbind();
   void deleteArray();
-  void linkVBO(const VBO &vbo, uint8 slot, uint8 n_components, GLenum type,
+  void linkVBO(VBO &vbo, uint8 slot, uint8 n_components, GLenum type,
```
Comment (Sephixum) [inline include/vao.hpp:23]: Could you share what the error was?
it's weird, since on my end it is fine and well, it is preferred to be flagged as "const".

**Comment by MiliAxe** [comment]
> Thank you for the hard work on your part. It is indeed appreciated.
> 
> Though there is a recommendation on class declaration part, for functions that do not change member variables please use "const" keyword after parameters parenthesis.
> 
> It is worth adding if function does not throw exceptions use keyword "noexcept" since it does in fact reduce executable size.
> 
> I would recommend taking a look at the [this](https://stackoverflow.com/questions/10787766/when-should-i-really-use-noexcept)

Thanks for the review. Noted. I will indeed add them ASAP.

File: include/vao.hpp (hunk 1 above)
Comment (MiliAxe) [reply include/vao.hpp]: I added this back to the method and seems like there are no errors this time

**Comment by MiliAxe** [comment]
I have added all the notes to the code. I think we are ready to merge the PR. We must consider implementing the "Vertex" struct/class soon as well.

**Comment by Sephixum** [comment]
> I have added all the notes to the code. I think we are ready to merge the PR. We must consider implementing the "Vertex" struct/class soon as well.

I reviewed changes, It is all fine, Vertex struct will be done in develop branch. Thank you.

**Pull request merged**